### ISSUE TITLE
Proposal: Packaging and Publishing in Python

### ISSUE THREAD

**Issue by serahkiburu** [carpentrycon/carpentryconhome-proposals#20]
### Session details

* Session type: Workshop 

* Keywords: packaging, publishing code, python 

* Permission to record this session: PENDING 

### Abstract

This workshop covers how to make your code more resusable and citable. Starting from an expectation of miscellaneous scripts and notebooks that are common among competent practitioners of python, who focus on their research analysis, we will cover how to organize your code into a pip-installable package with a live documentation website. 

The (alpha) lesson materials are located in the [Carpentries Incubator](https://carpentries-incubator.github.io/python-packaging-publishing/). This will be the second time this lesson is taught.  We will will collect feedback to improve the materials for others to use them. 

You should attend this workshop if:
- You identify with any of the [learner personas](https://carpentries-incubator.github.io/python-packaging-publishing/learner-profiles/index.html) for the lesson
- You have a rough idea how these tools work and want to firm up your knowledge to convince your research group to adopt more of these practices (eg by teaching the workshop locally later) 

### Personal details

* Name or pseudoname of the session lead: Sarah Brown

* Co-leads' names (we recommend involving 2 helpers/co-leads): PENDING

* Email or other ways to contact the session leads/co-leads: smb@sarahmbrown.org

* Country of residence and/or compatible Time Zones (provide options): US Eastern Time

* Would you like to present this multiple times, in other time zones: Could do both a morning and an evening to capture multiple time zones 

* Would you like to volunteer to be listed as a wrangler/host for your time zone: Contact me with more details  

### Is there any help you would like to invite from the community? Please provide below in bullet points.

* Looking for helpers (in the regular workshop sense)
* Looking for issue-documenters to note issues during the workshop in preparation for the following sprint (see #21 )

* Other comment: WRITE HERE

**Comment by serahkiburu** [comment]
Hi @brownsarahm, thanks for agreeing to run your breakout session remotely for CarpentryCon@Home. Please add details about your session to the sections with information pending above by editing the [issue text](https://github.com/carpentrycon/carpentryconhome-proposals/issues/20#issue-623192002).

**Comment by brownsarahm** [comment]
I don't have permission to edit the issue text

**Comment by serahkiburu** [comment]
you should be able to now, @brownsarahm

**Comment by brownsarahm** [comment]
Assigning doesn't give me permission to edit. I have triage level access and to edit I would need 'write' level access to this repo [per this summary of permissions](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization).  

I can quote reply or create my own issue.

Alternatively, though possibly a bigger change to your planned workflow, proposals could be made as PRs instead of issues. This would mean people use the PR review features to edit them collaboratively and then also make it so that the accepted proposals could be merged and then this repo would be able to serve as an archive of accepted proposals. 

PRs can be triaged in similar ways to issues with labels, in projects, etc. 

I'm happy to over the weekend submit separate PRs to:
1. update instructions 
2. submit my proposal

Just let me know your preference.

**Comment by serahkiburu** [comment]
I've given you and others write permissions @brownsarahm. However, feel free to open a new issue to submit your proposal, and I will close this one in case you do. Thanks.

**Comment by brownsarahm** [comment]
@ChristinaLK why did this get closed?

**Comment by ChristinaLK** [comment]
I assumed this was a duplicate of #21 - if not, my bad and I'll reopen!

**Comment by brownsarahm** [comment]
Nope, this is to run the actual workshop as a skillup, #21 is to have a sprint (preferably after) to improve the curriculum based on how the pilot goes

**Comment by ChristinaLK** [comment]
Got it - sorry about that!

**Comment by ousodaniel** [comment]
@brownsarahm thanks for your interest in regional hosting, check out the details [here](https://carpentries.org/blog/2020/06/call-for-carpentrycon-review-and-regional-hosts/) please. We'd love to see you sign up. Best

**Comment by tobyhodges** [comment]
I'd be happy to attend as a help/issue creator if the date/time of the session permits. US Eastern probably means I can only help with a morning session.

**Comment by brownsarahm** [comment]
@ChristinaLK Can Toby's help be considered in terms of scheduling? Should I edit the original and make that a comment to help?

**Comment by ChristinaLK** [comment]
Yes, we can take that into account. I'm sending out a scheduling form in just a bit and you can pick a time that works for both of you.

**Comment by serahkiburu** [comment]
Hi @brownsarahm, the organising committee reviewed this session proposal and had this to say:

>This workshop covers how to make your Python code more resusable and citable and the course material exist in Carpentries incubator. The proposal is very relevant for The Carpentries community. To help attendees decide if they would like to attend this session, we recommend the session leads to add clear pre-requisite and overall outcome of this event - for example, They have mentioned a list of personas - can they add a few examples in the abstract itself?

We are really excited for your session, and more details i.e. around scheduling will be shared with you in the coming days. Let me know if I can answer any questions or clarify anything for the time being. Thanks!
